src/MinimalDelegation.sol

```diff
@@ -191,18 +193,41 @@ contract MinimalDelegation is
     }
 
     /// @inheritdoc ERC1271
+    /// @dev WrappedSignature is used for both NestedTypedDataSign and NestedPersonalSign signatures.
+    ///      If the caller implementing ERC-1271 is considered safe, we verify the signature over the data directly
+    /// TypedDataSign signatures are of the form: abi.encode(bytes32, bytes(`signature ‖ APP_DOMAIN_SEPARATOR ‖ contentsHash ‖ contentsDescr ‖ uint16(contentsDescr.length)`))
+    /// No extra data is needed for NestedPersonalSign signatures.
     function isValidSignature(bytes32 data, bytes calldata wrappedSignature)
         public
         view
         override(ERC1271, IERC1271)
         returns (bytes4 result)
     {
+        // Per ERC-7739, return 0x77390001 for the sentinel hash value
+        unchecked {
+            if (wrappedSignature.length == uint256(0)) {
+                // Forces the compiler to optimize for smaller bytecode size.
+                if (uint256(data) == ~wrappedSignature.length / 0xffff * 0x7739) return 0x77390001;
+            }
+        }
+
         (bytes32 keyHash, bytes memory signature, bytes memory hookData) =
             abi.decode(wrappedSignature, (bytes32, bytes, bytes));
-        bytes32 digest = _hashTypedData(data.hashWithWrappedType());
 
         Key memory key = getKey(keyHash);
-        bool isValid = key.verify(digest, signature);
+
+        bool isValid;
+        if (erc1271CallerIsSafe[msg.sender]) {
+            isValid = key.verify(data, signature);
+        } else {
+            // We only support PersonalSign for ECDSA keys
+            if (signature.length == 65) {
+                isValid = _isValidNestedPersonalSignature(key, data, domainSeparator(), signature);
```

Comment:
nit: if shortening to sig in the name do it here as well for consistency?
so either _isValidNestedPersonalSignature and _isValidTypedDataSignature 

or
_isValidNestedPersonalSig and _isValidTypedDataSig